### PR TITLE
add: Orshot extension

### PR DESCRIPTION
**Pull request by rishimohan**
## Description

The PR adds Orshot Raycast extension which lets users generate website and tweet screenshots. After entering the link and hitting enter, it saves the screenshot on Desktop

## Screencast

Raycast 2025-09-20 at 22 35 19

Raycast 2025-09-20 at 22 35 57

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used in the `README` are located outside the metadata folder if they were not generated with our [metadata tool](https://developers.raycast.com/basics/prepare-an-extension-for-store#how-to-use-it)

### CODE REVIEW

File: extensions/orshot/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Orshot
```
[the remark this answers is not in the document]
Comment (rishimohan) [reply extensions/orshot/README.md:1]: Added the screens in metadata folder :)

File: extensions/orshot/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Orshot Changelog
+
+## [Initial Version] - September 20, 2025
```
Comment (xmok) [inline extensions/orshot/CHANGELOG.md]:
```suggestion
## [Initial Version] - {PR_MERGE_DATE}
```

File: extensions/orshot/src/create-website-screenshot.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  getPreferenceValues,
+  LaunchProps,
+} from "@raycast/api";
+import { writeFileSync } from "fs";
+import { join } from "path";
+import { homedir } from "os";
+
+interface Preferences {
+  apiKey?: string;
+}
```
Comment (xmok) [inline extensions/orshot/src/create-website-screenshot.ts]:
```suggestion
```
This is not needed.
`Preferences` type is auto-generated so you can write "Preferences" and it will work
(see auto-generated `raycast-env.d.ts`)

File: extensions/orshot/src/create-website-screenshot.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+interface Arguments {
+  websiteUrl: string;
+}
```
Comment (xmok) [inline extensions/orshot/src/create-website-screenshot.ts]:
```suggestion
```

File: extensions/orshot/src/create-website-screenshot.ts [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+interface ApiResponse {
+  data?: {
+    content?: string;
+  };
+  error?: string;
+}
+
+export default async function Command(
+  props: LaunchProps<{ arguments: Arguments }>,
```
Comment (xmok) [inline extensions/orshot/src/create-website-screenshot.ts]:
```suggestion
  props: LaunchProps<{ arguments: Arguments.CreateWebsiteScreenshot }>,
```
Arguments is also auto-generated.

File: extensions/orshot/src/create-tweet-screenshot.ts [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+interface ApiResponse {
+  data?: {
+    content?: string;
+  };
+  error?: string;
+}
+
+export default async function Command(
+  props: LaunchProps<{ arguments: Arguments }>,
```
Comment (xmok) [inline extensions/orshot/src/create-tweet-screenshot.ts]: See comments from other file.

**Review by xmok — CHANGES_REQUESTED** [review]
Hi @rishimohan:

Let's remove the extra files:
1. assets/extension-icon.png
2. raycast-env.d.ts

We're missing a few files:
1. .gitignore
2. .prettierrc

ONE
The actual generation seems to work ok. But the first 2 tweet I tried gave me an image with invalid tweet. Is it possible for the API to tell us if image is invalid and so don't generate image and let the user know?

TWO
instead of saving to Desktop, could we add a `Preference` letting user specify a directory of their choice?

Best.

**Comment by rishimohan** [comment]
@xmok Thanks for the quick feedback, I've updated the extension

Regarding the error with tweets, the issue is the provider we use caches tweets and right now it's tricky to accurately say if the tweet exists or not(in cases if it's deleted). Although I'll have a check on the API side which will provide a more relevant error text for this case

File: extensions/orshot/src/create-tweet-screenshot.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+import { homedir } from "os";
+
+interface ApiResponse {
+  data?: {
+    content?: string;
+  };
+  error?: string;
+}
+
+export default async function Command(props: LaunchProps) {
```
Comment (xmok) [inline extensions/orshot/src/create-tweet-screenshot.ts]:
```suggestion
export default async function Command(props: LaunchProps<{ arguments: Arguments.CreateTweetScreenshot}>) {
```
We need to type this properly.

File: extensions/orshot/src/create-tweet-screenshot.ts [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  await showToast({
+    style: Toast.Style.Animated,
+    title: "Generating Tweet Screenshot",
+    message: "Please wait...",
+  });
+
+  try {
+    const preferences = getPreferenceValues();
```
Comment (xmok) [inline extensions/orshot/src/create-tweet-screenshot.ts]:
```suggestion
    const preferences = getPreferenceValues<Preferences>();
```
The type is auto-generated but you still need to infer it.

File: extensions/orshot/src/create-website-screenshot.ts [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+import { homedir } from "os";
+
+interface ApiResponse {
+  data?: {
+    content?: string;
+  };
+  error?: string;
+}
+
+export default async function Command(props: LaunchProps) {
```
Comment (xmok) [inline extensions/orshot/src/create-website-screenshot.ts]:
```suggestion
export default async function Command(props: LaunchProps<{arguments: Arguments.CreateWebsiteScreenshot}>) {
```
Proper typing.

File: extensions/orshot/src/create-website-screenshot.ts [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  await showToast({
+    style: Toast.Style.Animated,
+    title: "Generating Website Screenshot",
+    message: "Please wait...",
+  });
+
+  try {
+    const preferences = getPreferenceValues();
```
Comment (xmok) [inline extensions/orshot/src/create-website-screenshot.ts]:
```suggestion
    const preferences = getPreferenceValues<Preferences>();
```

**Review by xmok — CHANGES_REQUESTED** [review]
re: the API errors, alright that makes sense!

In addition to the comments, you still need to update the date in your CHANGELOG to be "{PR_MERGE_DATE}".

Overall it looks good!

Have you tested it on Windows? Sometimes there are edge cases especially with `Arguments`. I'm not on my Windows device rn so can't test.

After that, LGTM!

**Comment by rishimohan** [comment]
@xmok Thanks, updated! I haven't tested it on Windows, I'm on Mac too, should I remove Windows support?

**Comment by xmok** [comment]
I'll test it on Windows in a few hours then we'll know. Let's leave it in for now.

File: extensions/orshot/package.json [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+  ],
+  "categories": [
+    "Web"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "create-tweet-screenshot",
+      "title": "Create Tweet Screenshot",
+      "subtitle": "Enter a link to tweet to create its screenshot",
```
Comment (xmok) [inline extensions/orshot/package.json]:
```suggestion
      "subtitle": "Orshot",
```

File: extensions/orshot/package.json [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+          "placeholder": "twitter.com/orshotapp/status/123456789",
+          "type": "text",
+          "required": true
+        }
+      ]
+    },
+    {
+      "name": "create-website-screenshot",
+      "title": "Create Website Screenshot",
+      "subtitle": "Enter a website url to create its screenshot",
```
Comment (xmok) [inline extensions/orshot/package.json]:
```suggestion
      "subtitle": "Orshot",
```

**Comment by rishimohan** [comment]
@xmok Thanks for the review! I've merged your commit suggestions about the subtitle issue. Let me know if any other changes are needed, thanks!

**Review by xmok — APPROVED** [review]
LGTM!
The Raycast Team is off this coming week so it might take 7+ days for this to be reviewed final time and merged.
Normally it does not take so long once approved.
Looking forward to seeing it in Store.
Best!

**Comment by rishimohan** [comment]
Ah, okay! Thanks again, looking for to see it merged for Store :)

**Comment by rishimohan** [comment]
Hi @xmok! just circling back, would be nice if you can help me make the extension live. Some users have pinged me about it :)

**Comment by rishimohan** [comment]
Hi Raycast Team! can anyone help with the PR review? please share if anything needs changing, looking forward to see it published. Thanks

**Comment by xmok** [comment]
Hey Rishi:
Nothing to worry about.
Some other 3 week old PRs got merged today so maybe yours will be done tomorrow or so.
Best!

**Comment by rishimohan** [comment]
Hi! pinging again for review! thanks 😅
